tools/src/tester/SchemaValidator.ts

```diff
@@ -35,6 +37,7 @@ export default class SchemaValidator {
     ajv_errors(this.ajv, { singleError: true })
     const schemas = spec.components?.schemas ?? {}
     for (const key in schemas) this.ajv.addSchema(schemas[key], `#/components/schemas/${key}`)
+    this.errors_parser = new AjvErrorsParser(this.ajv)
```

Comment:
This is an OO anti-pattern here. The error parser is really something that only applies to `this.ajv`, so it would be better to further subclass `AJV` into a `JsonSchemaValidator` (a generic JSON schema validator with error parsing), and make error parser a member of that, possibly hiding away more implementation details that are in `SchemaValidator` here and having its own tests. It would replace a few copy-pastes below. Can be done later of course.

Reply:
I also notice the same issue with AJV itself. That pattern is kinda already been broken by AJV when you have to do `ajv.errorsText(validate_func.errors)` instead of `validate_function.errorsText`. I can make a subclass for AJV to override its `errorsText` function but that means we will be locked to a specific JSON Schema draft (AJV vs AVJ2019 vs AJV2020).

Reply:
Actually it's not hard to subclass JsonSchemaValidator into JsonSchemaValidator2019 and JsonSchemaValidator2020.